### PR TITLE
add datadog metrics to meadow and playground

### PR DESCRIPTION
**Pull request by shmish111**

### CODE REVIEW

File: deployment/nixops/meadow.nix [hunk 1]
```diff
@@ -38,6 +39,19 @@ users.users.meadow = {
...
+  hostname = "${node.dns}";
+  api_key = datadogKey;
+  tags = [];
+};
+
+systemd.services.dogstatsd.serviceConfig = pkgs.lib.mkForce {
+  ExecStart = "${pkgs.dd-agent}/bin/dogstatsd";
+  User = "datadog";
+  Group = "datadog";
+};
```
Comment (michaelpj) [inline deployment/nixops/meadow.nix]: Maybe this should be in a module that you import in both configurations?

File: meadow/app/Webserver.hs [hunk 2]
```diff
@@ -79,9 +83,17 @@ app handlers _staticDir githubEndpoints config =
       simpleCorsResourcePolicy
         {corsRequestHeaders = ["content-type", "set-cookie"]}
 
+startStatsd :: MonadIO m => m Store
+startStatsd = liftIO $ do
+    store <- newStore
+    void $ forkStatsd defaultStatsdOptions store
+    pure store
```
Comment (michaelpj) [inline meadow/app/Webserver.hs]: Do we need to close this on shutdown, or is it a child process that will die gracefully?

File: plutus-playground-server/app/Webserver.hs [hunk 3]
```diff
@@ -79,9 +83,17 @@ app handlers _staticDir githubEndpoints config =
...
+
 run :: (MonadLogger m, MonadIO m) => Settings -> FilePath -> Config -> m ()
 run settings _staticDir config = do
+  store <- startStatsd
   githubEndpoints <- liftIO Auth.mkGithubEndpoints
   handlers <- PS.mkHandlers
+  appMonitor <- liftIO $ monitorEndpoints (Proxy @PA.API) store
   logInfoN "Starting webserver."
-  liftIO . runSettings settings $ app handlers _staticDir githubEndpoints config
+  liftIO . runSettings settings . appMonitor $ app handlers _staticDir githubEndpoints config
```
Comment (michaelpj) [inline plutus-playground-server/app/Webserver.hs]: Fair bunch of duplication between the two files but I'll let you deal with it when it's annoying.

File: deployment/nixops/default.nix [hunk 4]
```diff
@@ -4,9 +4,26 @@ let
   meadowMachine = import ./meadow.nix;
   machines = (plutus.pkgs.lib.importJSON ./machines.json);
   overlays = import ./overlays.nix;
-  playgroundConfig = plutus.pkgs.copyPathToStore ./playground.yaml;
+  secrets = (plutus.pkgs.lib.importJSON ./secrets.json);
```
Comment (michaelpj) [inline deployment/nixops/default.nix]: Can we get a note about how to use this?

File: meadow/app/Webserver.hs (hunk 2 above)
Comment (shmish111) [reply meadow/app/Webserver.hs]: according to the documentation there is no need to do anything

**Comment by shmish111** [comment]
@michaelpj for the rest of today I will be refactoring the nixops and terraform code so I was planning on pulling out a lot of duplication. In addition I will refresh the documentation in the PR for that as I might end up changing something.

**Comment by michaelpj** [comment]
Sounds good to me!

**Pull request merged**